src/com/google/enterprise/gsafeed/groups/Membership.java

```diff
@@ -32,11 +35,36 @@
 @XmlRootElement(name = "membership")
 public class Membership {
 
+  @XmlAttribute(name = "source")
+  @XmlJavaTypeAdapter(NormalizedStringAdapter.class)
+  protected String source;
   @XmlElement(required = true)
   protected Principal principal;
   @XmlElement(required = true)
   protected Members members;
 
+
+  /**
+   * Gets the value of the source attribute.
+   *
+   * @return possible object is {@link String}
+   */
+  public String getSource() {
+    return source;
+  }
+
+  /**
+   * Sets the value of the source attribute.
+   *
+   * @param value allowed object is {@link String}
+   * @return this object
+   */
+  public Membership setSource(String value) {
```

Comment:
Not sure what happened to the first comment I added. Is this attribute read-only? What happens if you send a groups feed with this attribute set differently from the global data source? Is it ignored, or used, or an error?

Reply:
It looks like the groupsource parameter in the request always wins. Groups feeds with mismatches or no source attributes at all in the membership elements came back in the downloaded groups file with the groupsource parameter value in the xml.